src/import/load-costume.js

```diff
@@ -30,76 +30,151 @@ const loadVector_ = function (costume, costumeAsset, runtime, rotationCenter, op
     return Promise.resolve(costume);
 };
 
-const loadBitmap_ = function (costume, costumeAsset, runtime, rotationCenter) {
+/**
+ * Return a promise to fetch a bitmap from storage and return it as a canvas
+ * If the costume has bitmapResolution 1, it will be converted to bitmapResolution 2 here (the standard for Scratch 3)
+ * If the costume has a text layer asset, which is a text part from Scratch 1.4, then this function
+ * will merge the two image assets. See the issue LLK/scratch-vm#672 for more information.
+ * @param {!object} costume - the Scratch costume object.
+ * @param {!Runtime} runtime - Scratch runtime, used to access the v2BitmapAdapter
+ * @param {?object} rotationCenter - optionally passed in coordinates for the center of rotation for the image. If
+ *     none is given, the rotation center of the costume will be set to the middle of the costume later on.
+ * @property {number} costume.bitmapResolution - the resolution scale for a bitmap costume.
+ * @returns {?Promise} - a promise which will resolve to an object {canvas, rotationCenter, assetMatchesBase},
+ *     or reject on error.
+ *     assetMatchesBase is true if the asset matches the base layer; false if it required adjustment
+ */
+const fetchBitmapCanvas_ = function (costume, runtime, rotationCenter) {
+    if (!costume || !costume.asset) {
+        return Promise.reject('Costume load failed. Assets were missing.');
+    }
     return new Promise((resolve, reject) => {
-        const imageElement = new Image();
+        const baseImageElement = new Image();
+        let textImageElement;
+
+        let loadedOne = false;
+
         const onError = function () {
             // eslint-disable-next-line no-use-before-define
             removeEventListeners();
-            reject('Image load failed');
+            reject('Costume load failed. Asset could not be read.');
         };
         const onLoad = function () {
-            // eslint-disable-next-line no-use-before-define
-            removeEventListeners();
-            resolve(imageElement);
+            if (loadedOne) {
```

Comment:
Maybe add a quick comment about what loadedOne represents. It was a little confusing at first. =D